### ISSUE TITLE
MapboxMobileEvents fails to build in Xcode 10–11 on CI

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-directions-swift#470]
[Xcode 10.2](https://circleci.com/gh/mapbox/mapbox-directions-swift/2056?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) and [Xcode 11.1](https://circleci.com/gh/mapbox/mapbox-directions-swift/2059?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) builds have been failing ever since 87ad66c9a50f19c0a833648c933fe7e2c4dbebbc upgraded from MapboxMobileEvents v0.10.2 to v0.10.4:

```
*** Building scheme "MapboxMobileEvents" in MapboxMobileEvents.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/distiller/project/Carthage/Checkouts/mapbox-events-ios/MapboxMobileEvents.xcodeproj -scheme MapboxMobileEvents -configuration Debug -derivedDataPath /Users/distiller/Library/Caches/org.carthage.CarthageKit/DerivedData/11.1_11A1027/mapbox-events-ios/v0.10.4 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/1b/gl7yt7ds26vcyr1pkgld6l040000gn/T/mapbox-events-ios SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/distiller/project/Carthage/Checkouts/mapbox-events-ios)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/1b/gl7yt7ds26vcyr1pkgld6l040000gn/T/carthage-xcodebuild.NSBup3.log
```

We need to grab the build log from one of these CI builds over SSH to understand the exact error. If it turns out to be fixed by mapbox/mapbox-events-ios#256, then we simply need to upgrade the dependency to v0.10.5. However, that fix was supposed to be for Xcode 12; I’m unsure why builds would fail in Xcode 10 and 11.

/ref https://github.com/mapbox/mapbox-directions-swift/pull/466#issuecomment-709050048
/cc @mapbox/navigation-ios @alfwatt @mr1sunshine

**Comment by 1ec5** [comment]
While we’re at it, we should remove Xcode 10 support in favor of Xcode 12 support: mapbox/mapbox-directions-swift#471.

**Comment by mr1sunshine** [comment]
@frederoni have you seen something like that before?

**Comment by avi-c** [comment]
I am `ssh`ed in to a failing session. The build error generated in MME is this:

`/MMEEventsManager.m:590:55: error: no visible @interface for 'CLLocation' declares the selector 'speedAccuracy'
                MMEEventKeySpeedAccuracy: @([location speedAccuracy]),
                                             ~~~~~~~~ ^~~~~~~~~~~~~
1 error generated.

I'll try updating our dependency to see if the update to v0.10.5 fixes it but it seems like a different issue. Based on the error is seems like a dependency on iOS 13 was (accidentally?) added by the Telemetry team.

**Comment by avi-c** [comment]
Here is the log of the failing build:

[carthage-xcodebuild.JYc9UQ.log](https://github.com/mapbox/mapbox-directions-swift/files/5430802/carthage-xcodebuild.JYc9UQ.log)

**Comment by alfwatt** [comment]
@avi-c you'll want the [0.10.5](https://github.com/mapbox/mapbox-events-ios/releases/tag/v0.10.5) build with [this fix](https://github.com/mapbox/mapbox-events-ios/commit/7731db5040d0ccbf10276c98d57ea7e62ca80586)

**Comment by avi-c** [comment]
Will be fixed with #474
